src/routes/safe/components/Settings/SpendingLimit/NewLimitModal/Review.tsx

```diff
@@ -0,0 +1,186 @@
+import { Button, Text } from '@gnosis.pm/safe-react-components'
+import React, { ReactElement, useMemo } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+
+import Block from 'src/components/layout/Block'
+import Col from 'src/components/layout/Col'
+import Row from 'src/components/layout/Row'
+import { getNetworkInfo } from 'src/config'
+import createTransaction from 'src/logic/safe/store/actions/createTransaction'
+import { SafeRecordProps, SpendingLimit } from 'src/logic/safe/store/models/safe'
+import {
+  addSpendingLimitBeneficiaryMultiSendTx,
+  adjustAmountToToken,
+  currentMinutes,
+  enableSpendingLimitModuleMultiSendTx,
+  setSpendingLimitMultiSendTx,
+  setSpendingLimitTx,
+  spendingLimitMultiSendTx,
+  SpendingLimitRow,
+} from 'src/logic/safe/utils/spendingLimits'
+import { MultiSendTx } from 'src/logic/safe/utils/upgradeSafe'
+import { makeToken, Token } from 'src/logic/tokens/store/model/token'
+import { fromTokenUnit, toTokenUnit } from 'src/logic/tokens/utils/humanReadableValue'
+import { sameAddress, ZERO_ADDRESS } from 'src/logic/wallets/ethAddresses'
+import { RESET_TIME_OPTIONS } from 'src/routes/safe/components/Settings/SpendingLimit/FormFields/ResetTime'
+import { AddressInfo, ResetTimeInfo, TokenInfo } from 'src/routes/safe/components/Settings/SpendingLimit/InfoDisplay'
+import Modal from 'src/routes/safe/components/Settings/SpendingLimit/Modal'
+import { useStyles } from 'src/routes/safe/components/Settings/SpendingLimit/style'
+import { safeParamAddressFromStateSelector, safeSpendingLimitsSelector } from 'src/logic/safe/store/selectors'
+
+import { ActionCallback, CREATE } from '.'
+
+const { nativeCoin } = getNetworkInfo()
+
+const useExistentSpendingLimit = ({
```

Comment:
I would move this to a hooks folder

Reply:
I'll leave it as it by now. It's not used in other places, and as with `useToken` what prevents us from making it a redux selector instead?